### ISSUE TITLE
Move PI profile images to official metadata schema

### ISSUE THREAD

**Issue by jeffbaumes** [microbiomedata/nmdc-metadata#314]
This is currently in the portal repo. Should we consider something like gravatar? Unsure if ORCID provides public access to a profile image but that might be ideal.

**Comment by dehays** [comment]
I don't see anywhere to include an image to a profile in ORCiD so I'd guess we can't get researcher images from there. Gravitar - maybe in cases where someone uses Wordpress or something else that caused them to add an avatar to Gravatar.

The PI profiles for the initial FICUS studies were very manually provided.  I think the best the schema can do here is provide an optional attribute for an image URL.  Thinking of the case where 10K+ studies are imported from NCBI - probably nothing there to set a profile image on import.

**Comment by jeffbaumes** [comment]
Agree, a URL would work fine and may be what we want to stick with at least near-term. Submitters could use the URL of their gravatar if they are savvy enough to do so I suppose. The important thing is to get it into the schema and not hard-coded in the client.

**Comment by jeffbaumes** [comment]
@jbeezley could you get a PR together to add an image URL added to the PI schema? We could just point current PI urls to our server for now, which is maybe not ideal but at least the PI info would be upstream. I expect the current mapping of PI to image URL could also be placed in the PR and used in the scripts that pull together that JSON from our current dataset.

@wdduncan could you point us to the code that creates PI JSON?

**Comment by wdduncan** [comment]
@jbeezley I am pulling the principle investigators name from the `contact` table in GOLD. To add this info to my ETL, I think I might have to add another table to the ETL ingest. Might be easier to have a call about how to best do it ...

**Comment by jbeezley** [comment]
We can't use a URL from our server because I store the data as a binary blob in the database.  It also depends on the uuid of the principal investigator row.  Perhaps we could base64 encode the data and put it in the provided study json?  For reference, the images used are stored in https://github.com/microbiomedata/nmdc-server/tree/master/nmdc_server/ingest/pis.

**Comment by jbeezley** [comment]
> I don't see anywhere to include an image to a profile in ORCiD so I'd guess we can't get researcher images from there. Gravitar - maybe in cases where someone uses Wordpress or something else that caused them to add an avatar to Gravatar.

Gravatar would be a good alternative, but we don't necessarily get email addresses from ORCiD's.  I checked some of our existing PI's and they don't make email public.

**Comment by jeffbaumes** [comment]
Would it be reasonable to support both with allowing either a URL or [data URI](https://css-tricks.com/data-uris/)? As long as it is validated as one or the other, we could safely pass this through to the `img` `src` attribute and it would work in either case.

**Comment by dwinston** [comment]
@jeffbaumes I'd rather there by only one (external URL) versus two modes. It should be no more difficult to supply a URL for a profile image than for a data object.

On the topic of offering use of a Gravar image, I think that can be convenient, but should be opt-in somehow. Some PIs may have gravatar images from a decade ago that they have forgotten about and would prefer not to use.

**Comment by jeffbaumes** [comment]
You could think of my suggestion as actually only one thing: provide a [URI](https://en.wikipedia.org/wiki/Uniform_Resource_Identifier) to an image. It just happens there are multiple types of URI we could support fairly easily. They could be handled identically start to finish, with no extra logic anywhere to support them other than a more complex regex for validation, so I'm not seeing much downside.

I'd be ok with URL, and savvy PIs could find and use their gravatar URL so in that sense it would be opt-in. The main extra-work-for-us for URL-only is that we would need to decide where to host the current PI images. They need to be at static, stable URLs.

**Comment by wdduncan** [comment]
All these ideas are fine with me, but where in the ETL do we insert the URL/URI? I can do it on my end by simply having a file that gives the URI for each investigator in the `contact` table. However, this won't work for investigators that not registered in the GOLD database.

**Comment by jbeezley** [comment]
We can make it optional and on the portal show a placeholder image.  A lot of these questions on what to make required (#310) depend on features needed for the portal and what we are willing to leave blank.

**Comment by dehays** [comment]
Moving to nmdc-schema to add optional pi_image_url slot 

For the studies we currently have (12 FICUS) - will need to put image files on NERSC and manually set them in metadata - there is no standard source for these images so I don't see a way the GOLD ETL can set these

@wdduncan Question on implementation - an image_url on the person entity seems correct to me.  Then study would refer to the PI (a person) image_url attribute.  I think this is what you are doing with the principal_investigator_name on study. Does this make sense?
